plugins/modules/kinesis_stream.py

```diff
@@ -361,6 +361,8 @@ def find_stream(client, stream_name, check_mode=False):
                 )
                 shards.extend(results.pop('Shards'))
                 has_more_shards = results['HasMoreShards']
+                if has_more_shards:
```

Comment:
ran into an index out of range error when creating new kinesis streams that don't have shards assigned to them yet so added in this check